x-pack/metricbeat/module/aws/utils_test.go

```diff
@@ -169,3 +170,82 @@ func TestGetMetricDataResults(t *testing.T) {
 	assert.Equal(t, label4, *getMetricDataResults[3].Label)
 	assert.Equal(t, 0.0, getMetricDataResults[3].Values[0])
 }
+
+func TestCheckTimestampInArray(t *testing.T) {
+	timestamp1 := time.Now()
+	timestamp2 := timestamp1.Add(5 * time.Minute)
+	timestamp3 := timestamp1.Add(10 * time.Minute)
+
+	timestampArray := []time.Time{timestamp1, timestamp2}
+	exists1, index1 := CheckTimestampInArray(timestamp1, timestampArray)
+	assert.True(t, exists1)
+	assert.Equal(t, 0, index1)
+
+	exists2, index2 := CheckTimestampInArray(timestamp3, timestampArray)
+	assert.False(t, exists2)
+	assert.Equal(t, -1, index2)
+}
+
+func TestFindTimestamp(t *testing.T) {
+	timestamp1 := time.Now()
+	timestamp2 := timestamp1.Add(5 * time.Minute)
+	getMetricDataResults1 := []cloudwatch.MetricDataResult{
```

Comment:
I think this could be nicely converted to a table driven test: https://github.com/golang/go/wiki/TableDrivenTests Can also be done in a follow up PR.

Reply:
Great suggestion! Thanks!